HHS/Head-Start-TTADP
# [TTAHUB-1029] fix editing experience

## Description of change
- Changes to URL structure to make it more meaningful when editing multiple goals, updated edit menu to use this new structure. Visual changes to the goal form to make the fact that multiple goals are being edited more obvious (Goal X, Y now shows in the title)
- New hook to read URL parameters into state
- Fix roll up logic on the create goals form and the RTR so that the data across all the goals and objectives is displayed
- Also had to create logic to roll up other entity goals

## How to test
Create a multi-grant single recipient goal. Edit it from the RTR table. Use it on an approved AR. Check the RTR again, make sure that you can still edit it and that the info is properly displayed in the table.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1029


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
